### PR TITLE
Better error message when (self_build_stubs_archive ...) and (c_names ...) or (cxx_names ...) used simultaneously

### PR DESCRIPTION
**Pull request by nojb**
See #1306. Suggestions for better wording welcome.

### CODE REVIEW

File: src/dune_file.ml [hunk 1]
```diff
@@ -1014,6 +1014,23 @@ module Library = struct
             It is inherited from the virtual library."
        | _, _, _ -> ()
        end;
+       let self_build_stubs_archive =
+         let loc, self_build_stubs_archive = self_build_stubs_archive in
+         let err =
+           match c_names, cxx_names, self_build_stubs_archive with
+           | _, _, None -> None
+           | (_ :: _), _, Some _ -> Some ("c_names", loc)
+           | _, (_ :: _), Some _ -> Some ("cxx_names", loc)
```
Comment (rgrinberg) [inline src/dune_file.ml]: Why do we return the loc? It's the same in all clauses.

Comment (nojb) [reply src/dune_file.ml]: Indeed, it was left-over from previous code. Fixed.

**Review by rgrinberg — APPROVED** [review]

**Comment by diml** [comment]
The CI failure is probably the same as #1355, we might need to add a `touch` in this test as well.

**Review by ghost — APPROVED** [review]

**Comment by diml** [comment]
BTW, we could also support having both if we wanted. It should be possible to attach several archives to  a .cm[x]a.

**Comment by diml** [comment]
The travis failure should be fixed after a rebase

**Comment by diml** [comment]
Ok, feel free to merge :)

**Comment by nojb** [comment]
Done. Thanks! 😄

**Pull request merged**